storm-server/src/main/java/org/apache/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -248,8 +252,17 @@ private void scheduleTopology(TopologyDetails td, Cluster cluster, final User to
     }
 
     @VisibleForTesting
-    public Set<String> getEvictedTopologies() {
-        return this.evictedTopologies;
+    public Set<String> getEvictedTopologies() throws Exception {
+        if (forTest) {
+            return this.evictedTopologies;
+        } else {
+            throw new Exception("Topology eviction check is only provided for test purposes");
```

Comment:
This can be `UnsupportedOperationException`

Reply:
Fixed. Thank you.